buildrunner/config/models.py

```diff
@@ -161,12 +161,20 @@ def validate_steps(cls, vals) -> None:
         if not vals:
             raise ValueError('The "steps" configuration was not provided')
 
-        # Checks to see if there is a mutli-platform build step in the config
+        # Checks steps for mutli-platform or secrets
         has_multi_platform_build = False
+        has_secrets = False
         for step in vals.values():
             has_multi_platform_build = (
                 has_multi_platform_build or step.is_multi_platform()
             )
+            has_secrets = has_secrets or step.has_secrets()
+
+        if has_secrets:
+            if info.data.get("use_legacy_builder"):
```

Comment:
does this work properly with `platforms` set which infers `use_legacy_builder` as false?

Reply:
It defaults to `True` whether it is explicitly set or not.

Reply:
I am making a code update so that platforms doesn't need to set `use-legacy-builder`

Reply:
Add better validation and made it so that multi-platform builds don't have to specify use_legacy_builder.